src/index.ts

```diff
@@ -14,4 +15,21 @@ if (typeof afterEach === 'function' && !process.env.RNTL_SKIP_AUTO_CLEANUP) {
   });
 }
 
+if (
+  typeof beforeAll === 'function' &&
+  typeof afterAll === 'function' &&
+  !process.env.RNTL_SKIP_AUTO_CLEANUP
```

Comment:
Should we check `RNTL_SKIP_AUTO_CLEANUP` flag? This does not look like it's strictly related to autocleanup feature.

Reply:
If you don't do that, you have nested `beforeAll` in the test that tests for cleanup^^' and that breaks it. Current implementation is the same as in RTL. And alternative would be to break out the `index.ts` in something that re-exports pure, + two files once for the clean up, once for the ACT thingy, and in the test file, only import the other one. Or add another custom global just to ignore it in the tests.

Reply:
Can we structure it more like RTL, so that it's easier to sync code in the future:

```
if (typeof process === 'undefined' || !process.env?.RTL_SKIP_AUTO_CLEANUP) {
  // If we're running in a test runner that supports afterEach
  // then we'll automatically run cleanup afterEach test
  // this ensures that tests run in isolation from each other
  // if you don't like this then either import the `pure` module
  // or set the RNTL_SKIP_AUTO_CLEANUP env variable to 'true'.
  if (typeof afterEach === 'function') {
    // eslint-disable-next-line no-undef
    afterEach(async () => {
      await flushMicroTasks();
      cleanup();
    });
  }

  if (typeof beforeAll === 'function' && typeof afterAll === 'function') {
    // This matches the behavior of React < 18.
    let previousIsReactActEnvironment = getIsReactActEnvironment();
    beforeAll(() => {
      previousIsReactActEnvironment = getIsReactActEnvironment();
      setReactActEnvironment(true);
    });
  
    afterAll(() => {
      setReactActEnvironment(previousIsReactActEnvironment);
    });
  }
}
```